sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/kafka/BindingServicePropertiesBeanPostProcessor.java

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.spring.cloud.autoconfigure.kafka;
+
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.config.BeanPostProcessor;
+import org.springframework.cloud.stream.config.BinderProperties;
+import org.springframework.cloud.stream.config.BindingServiceProperties;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * {@link BeanPostProcessor} for {@link BindingServiceProperties} to support Azure Identity for Kafka Spring Cloud Stream Binder.
+ * This BeanPostProcessor will put the {@link AzureKafkaSpringCloudStreamConfiguration} into Kafka binder's context.
+ *
+ * @since 4.4.0
+ */
+public class BindingServicePropertiesBeanPostProcessor implements BeanPostProcessor {
+
+    static final String SPRING_MAIN_SOURCES_PROPERTY = "spring.main.sources";
+    private static final String KAKFA_BINDER_DEFAULT_NAME = "kafka";
+    private static final String KAKFA_BINDER_TYPE = "kafka";
+
+    @Override
+    public Object postProcessBeforeInitialization(Object bean, String beanName) throws BeansException {
+        if (bean instanceof BindingServiceProperties) {
+            BindingServiceProperties bindingServiceProperties = (BindingServiceProperties) bean;
+            if (bindingServiceProperties.getBinders().isEmpty()) {
+                BinderProperties kafkaBinderSourceProperty = new BinderProperties();
+                kafkaBinderSourceProperty.setType(KAKFA_BINDER_TYPE);
```

Comment:
Do we need this setting binder type?

Reply:
YEAH, this can be ignored since we have set the binder name as "kafka"